Backpack/FloatingNotification/Classes/BPKFloatingNotification.swift

```diff
@@ -0,0 +1,203 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import UIKit
+
+public final class BPKFloatingNotification: UIView {
+    
+    private struct Constants {
+        static let buttonHeight: CGFloat = 20.0
+        static let maxWidth: CGFloat = 400.0
+        static let startBottomConstraint: CGFloat = 30.0
+    }
+    
+    private var onButtonTap: (() -> Void)?
+    private var onDismissal: (() -> Void)?
+    
+    private var bottomConstraint = NSLayoutConstraint()
+    private let animator = FloatingNotificationAnimator()
+    private var notificationsQueue = [FloatingNotificationViewModel]()
+    
+    // MARK: Subviews
+    private let stackView: UIStackView = {
+        let result = UIStackView()
```

Comment:
nit: could we rename this `result` into `stack`, `iconView`, `label`, and such?

Reply:
Yeah sure can do